examples/e-commerce/src/app/snippet/snippet.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-snippet',
+  template: `<span class="ais-Snippet" [innerHtml]="content"></span>`,
+})
+export class Snippet {
+  @Input() attribute: string;
+  @Input() hit: { _snippetResult?: {}; label?: string; highlighted?: string };
+  @Input() tagName: string = 'em';
+
+  get content() {
+    if (this.hit.hasOwnProperty('_snippetResult')) {
+      const attributeHighlighted = this.hit._snippetResult[this.attribute];
+
+      // check that the attributeHighlighted is a string
+      if (
+        attributeHighlighted !== undefined &&
+        typeof attributeHighlighted.value === 'string'
+      ) {
+        return this.replaceWithTagName(attributeHighlighted.value);
+      }
+    }
+
+    const fallback = this.hit[this.attribute];
+    if (!fallback) {
```

Comment:
This shouldn't happen in our demo, I think it's safe to remove it.

Reply:
fe05cbd